### PR TITLE
Using PapaParse inside a worker

### LINKED ISSUE #188: Using PapaParse inside a worker
I have the problem that the parser always wants to post the results using postMessage because I'm using PapaParse inside a worker.  I guess it's because it thinks it is one of the workers the parser spawned. 

Is there any kind of global where I can deactivate all the worker behaviour?

### PR DESCRIPTION
**Pull request by benekastah**
Fixes #188

### CODE REVIEW

**Comment by mholt** [comment]
Cool, I'll take a look at this soon!

**Comment by benekastah** [comment]
@mholt If you can look at this soon, I'll fix the conflicts.

**Comment by mholt** [comment]
@benekastah Thanks for the push :smile: Looked at it just now. Looks fine, seems like a logical way to do it. Haven't tried it but if the tests pass, let's merge it in!

**Comment by benekastah** [comment]
Cool, I'll fix up the conflicts and make sure the tests pass then.

**Comment by bluej100** [comment]
I think we might want to separate IS_WORKER into two separate flags: one for whether we are in a worker and one for whether the current parse was called directly or through a posted message. For example, the IS_WORKER checks for synchronous XMLHttpRequest and FileReader usage should probably continue to be based on the !global.document check alone. My suggestion would be that instead in workerThreadReceivedMessage we add an isWorker property to the config and check that in chunkError, etc.

**Comment by bluej100** [comment]
Certainly it's unlikely that a single worker would do parses both directly and indirectly, so the second flag could reasonably be a constant as well.

**Comment by benekastah** [comment]
Ok, I should have fixed the tests. @bluej100 I think I know what you mean, I'll make that modification now and see if it's what you had in mind.

**Comment by mholt** [comment]
@bluej100 As you know, it's Pioneer Day here in Utah so I won't be online most of today. If you like what you see, feel free to merge the PR when it's ready.

**Comment by bluej100** [comment]
Sir.

**Comment by benekastah** [comment]
@bluej100 Ok, what I did is slightly different than what you described, but I think it captures the spirit of what you wanted. I couldn't just add an `isWorker` property to the config in `workerThreadReceivedMessage` because the main problem I was trying to prevent was that papaparse was setting `workerThreadReceivedMessage` as the `onmessage` callback even if papaparse wasn't the worker's owner. So I changed the query string to be "papaworker" rather than just "worker" as the signal the papaparse spawned it and is the owner.

File: tests/test-cases.js [hunk 1]
```diff
@@ -1458,6 +1458,49 @@ var CUSTOM_TESTS = [
 				}
 			});
 		}
+	},
+	{
+		description: "Should not assume we own the worker unless papaworker is in the search string",
+		expected: [false, true, true, true, true],
+		run: function(callback) {
+			if (typeof Worker === 'undefined') {
```
Comment (bluej100) [inline tests/test-cases.js]: Perhaps you could use the disabled property here, as used with FILES_ENABLED?

File: papaparse.js [hunk 2]
```diff
@@ -1395,3 +1400,5 @@
 		return typeof func === 'function';
 	}
 })(typeof window !== 'undefined' ? window : this);
+
+// vim: noet
```
Comment (bluej100) [inline papaparse.js]: Mind removing this?

File: tests/test-cases.js [hunk 3]
```diff
@@ -1458,6 +1458,49 @@ var CUSTOM_TESTS = [
...
+				}
+			};
+
+			searchStrings.forEach(function (searchString, idx) {
+				var w = new Worker('../papaparse.js' + searchString);
+				// Give it .5s to do something
+				var timeout = setTimeout(function () {
+					w.terminate();
+					setResult(idx, false);
+				}, 500);
```
Comment (bluej100) [inline tests/test-cases.js]: You could simplify this a bit by initializing to false and having a single timeout with callback(results), if you like.

**Comment by bluej100** [comment]
Looks good to me. Left a couple suggestions on the test. Thanks for your work on this, @benekastah .

**Comment by benekastah** [comment]
Cool. Addressed all your comments.

**Comment by benekastah** [comment]
I changed `WORKERS_SUPPORTED` per your suggestion (but I don't find your comment anymore...)

**Comment by bluej100** [comment]
Merged. Thanks a bunch!

**Comment by benekastah** [comment]
Thank you!

**Comment by benekastah** [comment]
Is there a way I can get notified when this PR makes it into a new version?

**Comment by mholt** [comment]
As soon as I can get around to it, I'll try to remember. Been really busy with other projects and travel.

**Comment by mholt** [comment]
@benekastah Deployed in version 4.1.2.

**Comment by benekastah** [comment]
Thanks @mholt!

**Pull request merged**